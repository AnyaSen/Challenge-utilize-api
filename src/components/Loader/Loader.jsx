import React from 'react';
import Styles from './Loader.module.scss';

import loader from '../../assets/img/play-button.svg';

export default function Loader() {
	return (
		<div data-testid='Loader' className={Styles.LoaderContainer}>
			<img src={loader} alt='loader icon' />
			<h3>Loading...</h3>
		</div>
	);
}
