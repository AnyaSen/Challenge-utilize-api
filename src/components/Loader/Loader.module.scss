@import '../../assets/scss/variables';
.LoaderContainer {
	display: flex;
	flex-direction: column;
	align-items: center;
	img {
		margin-bottom: $space-2;
	}
}
