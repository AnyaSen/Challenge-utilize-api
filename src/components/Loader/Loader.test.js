import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Loader from './Loader';

describe('<Loader />', () => {
	afterAll(cleanup);

	test('Should contain the correct image & text', () => {
		const { getByText, getByAltText, container } = render(<Loader />);

		const loadingMessage = 'Loading...';

		const loadingText = getByText(loadingMessage);
		const loaderIcon = getByAltText('loader icon');

		expect(loadingText).toBeTruthy();
		expect(loaderIcon.src).toContain('play-button.svg');

		expect(container.firstChild).toMatchSnapshot();
	});
});
