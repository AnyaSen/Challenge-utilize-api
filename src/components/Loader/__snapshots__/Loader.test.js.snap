// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Should contain the correct image & text 1`] = `
<div
  class="LoaderContainer"
  data-testid="Loader"
>
  <img
    alt="loader icon"
    src="play-button.svg"
  />
  <h3>
    Loading...
  </h3>
</div>
`;
